test/functional/lua/buffer_updates_spec.lua

```diff
@@ -449,6 +449,19 @@ describe('lua buffer event callbacks: on_lines', function()
     eq({ 'bytes', 2, 5, 0, 0, 0, 4, 0, 40, 0, 10, 10 }, api.nvim_get_var('qf_on_bytes'))
     eq({ 'lines', 2, 6, 0, 5, 1, 42 }, api.nvim_get_var('qf_on_lines'))
   end)
+
+  it('single-line visual block insert should not trigger extra on_lines #22009', function()
```

Comment:
should this test be in `test/functional/api/buffer_updates_spec.lua` ? buf updates not a lua-only feature.

this test file is only for the lua-only parts of buffer updates.

Reply:
I guess no .. see describe `describe('lua buffer event callbacks: on_lines', function()` the on_lines cb test case all in this block.